File: 백준 문제 풀이 모음/class 3/5430.py
import sys

T = int(sys.stdin.readline())

def fun_R (array) :
    if len(array) == 0 :
        return []
    else :
        arr = array[::-1]
    return arr


def fun_D (array) :
    
    array.pop(0)
    return array

for _ in range(T) :
    p = list(sys.stdin.readline().rstrip())
    n = int(sys.stdin.readline())
    array = sys.stdin.readline().rstrip()
    if n == 0 :
        array = []
    else :
        array = array[1:-1].split(',')
    
    for i in p :
        if i == 'R' :
            array = fun_R(array)          
        elif i == 'D' :
            if n == 0 or len(array) == 0 :
                array = 'error'
                
                break
            else :
                array = fun_D(array)

    if array == 'error' :
        print('error')
    else :
        result = []
        for i in array :
            result.append(int(i))
        print(result)

File: 백준 문제 풀이 모음/kotlin/class 3/5430.kt
package class3

val sb = StringBuilder()

fun main() = with(System.`in`.bufferedReader()) {
    val T = readLine().toInt()
    repeat(T) {
        val arr = readLine().toCharArray() // arr = RDD
        val n = readLine().toInt()
        val input = readLine().removeSurrounding("[", "]").split(',').toMutableList()

        val deque = ArrayDeque<Int>()

        input.forEach {
            if (!it.equals("")) {
                deque.add(it.toInt())
            }
        }
        var err = 0
        var r_count = 0
        for (it in 0 until arr.size) {
            if (arr[it].equals('R')) {
                r_count ++
            } else if(arr[it].equals('D')) {
                if (deque.size < 1) {
                    err = 1
                    println("error")
                    break
                } else {
                    if (r_count % 2 == 0) {
                        deque.removeFirst()
                    } else {
                        deque.removeLast()
                    }
                }
            }
        }
        if (err == 0) {
          if (r_count % 2 != 0) {
              deque.reverse()
              println(deque.toString().replace(", ", ","))

          } else {
              println(deque.toString().replace(", ", ","))

          }
        }
    }
    print(sb.toString())


}